test/behavior/zon.zig

```diff
@@ -0,0 +1,268 @@
+        const found_slice: []const u8 = @import("zon/slice-abc.zon");
+        try expectEqualSlices(u8, expected_slice, found_slice);
+
+        const expected_array: [3]u8 = .{ 'a', 'b', 'c' };
+        const found_array: [3]u8 = @import("zon/slice-abc.zon");
+        try expectEqual(expected_array, found_array);
+
+        const T = struct { u8, u8, u8 };
+        const expected_tuple: T = .{ 'a', 'b', 'c' };
+        const found_tuple: T = @import("zon/slice-abc.zon");
+        try expectEqual(expected_tuple, found_tuple);
+    }
+}
+
+test "string literals" {
+    // const foo: [3]u8 = "foo".*;
+    // const bar: []const u8 = &foo;
+    try expectEqualSlices(u8, "abc", @import("zon/abc.zon"));
+    try expectEqualSlices(u8, "ab\\c", @import("zon/abc-escaped.zon"));
+    const zero_terminated: [:0]const u8 = @import("zon/abc.zon");
+    try expectEqualDeep(zero_terminated, "abc");
+    try expectEqualStrings(
+        \\Hello, world!
+        \\This is a multiline string!
+        \\ There are no escapes, we can, for example, include \n in the string
+    , @import("zon/multiline_string.zon"));
+    try expectEqualStrings("a\nb\x00c", @import("zon/string_embedded_null.zon"));
+}
+
+test "enum literals" {
+    const Enum = enum {
+        foo,
+        bar,
+        baz,
+        @"0\na",
+    };
+    try expectEqual(Enum.foo, @import("zon/foo.zon"));
+    try expectEqual(Enum.@"0\na", @import("zon/escaped_enum.zon"));
+}
+
+test "int" {
+    const expected = .{
+        // Test various numbers and types
+        @as(u8, 10),
+        @as(i16, 24),
+        @as(i14, -4),
+        @as(i32, -123),
+
+        // Test limits
+        @as(i8, 127),
+        @as(i8, -128),
+
+        // Test characters
+        @as(u8, 'a'),
+        @as(u8, 'z'),
+
+        // Test big integers
+        @as(u65, 36893488147419103231),
+        @as(u65, 36893488147419103231),
+        @as(i128, -18446744073709551615), // Only a big int due to negation
+        @as(i128, -9223372036854775809), // Only a big int due to negation
+
+        // Test big integer limits
+        @as(i66, 36893488147419103231),
+        @as(i66, -36893488147419103232),
+
+        // Test parsing whole number floats as integers
+        @as(i8, -1),
+        @as(i8, 123),
+
+        // Test non-decimal integers
+        @as(i16, 0xff),
+        @as(i16, -0xff),
+        @as(i16, 0o77),
+        @as(i16, -0o77),
+        @as(i16, 0b11),
+        @as(i16, -0b11),
+
+        // Test non-decimal big integers
+        @as(u65, 0x1ffffffffffffffff),
+        @as(i66, 0x1ffffffffffffffff),
+        @as(i66, -0x1ffffffffffffffff),
+        @as(u65, 0x1ffffffffffffffff),
+        @as(i66, 0x1ffffffffffffffff),
+        @as(i66, -0x1ffffffffffffffff),
+        @as(u65, 0x1ffffffffffffffff),
+        @as(i66, 0x1ffffffffffffffff),
+        @as(i66, -0x1ffffffffffffffff),
+    };
+    const actual: @TypeOf(expected) = @import("zon/ints.zon");
```

Comment:
This coercion is a little weird because `@TypeOf(expected)` is a tuple full of `comptime` fields -- so if the coercion here succeeds, the field *values* matched, so the test below can never fail! You might need to explicitly write out the tuple type, i.e. `struct { u8, i16, i14, ... }`.